### ISSUE TITLE
[mlir] Lit tests fail after BF16 ABI change

### ISSUE THREAD

**Issue by phoebewang** [llvm/llvm-project#57042]
I had a patch https://github.com/llvm/llvm-project/commit/c7ec6e19d5446a448f888b33f66316cf2ec6ecae to make the `bfloat` type to follow X86 psABI. It seems the change makes MLIR tests failed again:
```
Failed Tests (2):
  MLIR :: Integration/Dialect/SparseTensor/CPU/dense_output_bf16.mlir
  MLIR :: Integration/Dialect/SparseTensor/CPU/sparse_sum_bf16.mlir
```
I suppose it is the same reason as #55992. cc @joker-eph @aartbik do I need to disable them temporarily, or you have a quick fix? Please notice, I'm planning to backport the change to LLVM 15.0 too.

**Comment by aartbik** [comment]
The ABI issue was supposed to be fixed by https://reviews.llvm.org/D128018 which passes the value through memory (by reference rather than by value). If it breaks again I really would like to know why and not disable the tests anymore.

**Comment by aartbik** [comment]
Discussed this with  @joker-eph. For the sake of progress, I am okay disabling the tests and investigating what happened "offline".

**Comment by phoebewang** [comment]
Thanks @joker-eph and @aartbik . I think the problem is in the runtimes that still expect BF16 type passed as `uint16_t`. I think changing it to `__bf16` or even `float` for X86 targets should solve the problem.
https://github.com/llvm/llvm-project/blob/main/mlir/lib/ExecutionEngine/Float16bits.cpp#L149-L175

**Comment by phoebewang** [comment]
FYI. This is what I changed in compiler-rt https://reviews.llvm.org/D131147

**Comment by aartbik** [comment]
Thanks @d0k for the fix in f695554a2a55
